core-bundle/contao/library/Contao/Controller.php

```diff
@@ -594,6 +595,16 @@ public static function getContentElement($intId, $strColumn='main')
 		}
 		else
 		{
+			if (\is_array($contentElementReference?->attributes['classes'] ?? null))
+			{
+				if ($objRow instanceof Model && Registry::getInstance()->isRegistered($objRow))
```

Comment:
`$objRow instanceof Model` is always true here.
And we should not check the registry as we also should also not modify models that got passed to the `getContentElement()` method, even if they are not registered.

I’d add the line `$objRow = $objRow->cloneOriginal();` before line 572 where we already modify `$objRow->typePrefix`.

Reply:
> $objRow instanceof Model is always true here.

That is not true. It is an object, but it can also be a `Contao\Database\Result` or a Collection actually. See https://github.com/contao/contao/blob/5.x/core-bundle/contao/elements/ContentElement.php#L147

Reply:
> It is an object, but it can also be a `Contao\Database\Result` or a Collection actually.

Then line 556 would already fail because of the `Model` type hint: https://github.com/contao/contao/blob/da1b9ea8998ca8547a8041d3560cb3efbdaee755/core-bundle/contao/library/Contao/Controller.php#L383 https://github.com/contao/contao/blob/da1b9ea8998ca8547a8041d3560cb3efbdaee755/core-bundle/contao/library/Contao/Controller.php#L745

Reply:
changed in 39f94fe57c158c949f6c3a75fd7404d82478dd9e